metadata-ingestion/src/datahub/ingestion/source/iceberg/iceberg_profiler.py

```diff
@@ -0,0 +1,117 @@
+
+    def _aggregateCounts(self, aggregatedCount: Dict[str, int], fieldPaths: Dict[int, str], manifestCounts: Dict[int, int]):
+        for field_id, count in manifestCounts.items():
+            fieldPath = fieldPaths.get(field_id)
+            if not fieldPath:
+                raise RuntimeError(f"Failed to find fieldPath for field_id {field_id}")
+            agg = aggregatedCount.get(fieldPath) or 0
+            aggregatedCount[fieldPath] = agg + count
+
+    def _aggregateBounds(self, schema: Schema, aggregator: Callable, aggregatedValues: Dict[str, Any], fieldPaths: Dict[int, str], manifestValues: Dict[int, Any]):
+        for field_id, valueEncoded in manifestValues.items():
+            field = schema.find_field(field_id)
+            valueDecoded = Conversions.from_byte_buffer(field.type, valueEncoded)
+            if valueDecoded:
+                fieldPath = fieldPaths.get(field_id)
+                if not fieldPath:
+                    raise RuntimeError(f"Failed to find fieldPath for field_id {field_id}")
+                agg = aggregatedValues.get(fieldPath)
+                aggregatedValues[fieldPath] = aggregator(agg, valueDecoded) if agg else valueDecoded
+
+    def profileTable(self, env: str, dataset_name: str, table: Table, schemaMetadata: SchemaMetadata) -> MetadataWorkUnit:
+        rowCount = int(table.current_snapshot().summary["total-records"])
+        dataset_profile = DatasetProfileClass(
+            timestampMillis=get_sys_time(),
+            rowCount=rowCount,
+            columnCount=len(schemaMetadata.fields),
+            fieldProfiles=[]
+            # columnCount=len(table.schema().columns())
+        )
+
+        fieldPaths: dict[int, str] = table.schema()._id_to_name #dict[int, str] = {v: k for k, v in index_by_name(table.schema()).items()}
+        current_snapshot: Snapshot = table.current_snapshot()
+        totalCount = 0
+        nullCounts = {}
+        valueCounts = {}
+        minBounds = {}
+        maxBounds = {}
+        manifest: ManifestFile
+        try:
+            for manifest in current_snapshot.manifests:
+                manifest_input_file = FileSystemInputFile.from_location(manifest.manifest_path, table.ops.conf)
+                manifest_reader = ManifestReader.read(manifest_input_file)
+                dataFile: DataFile
+                for dataFile in manifest_reader.iterator():
+                    self._aggregateCounts(nullCounts, fieldPaths, dataFile.null_value_counts())
+                    self._aggregateCounts(valueCounts, fieldPaths, dataFile.value_counts())
+                    self._aggregateBounds(table.schema(), min, minBounds, fieldPaths, dataFile.lower_bounds())
+                    self._aggregateBounds(table.schema(), max, maxBounds, fieldPaths, dataFile.upper_bounds())
+                    totalCount += dataFile.record_count()
+        # TODO Work on error handling to provide better feedback.  Iceberg exceptions are weak...
+        except FileSystemNotFound as e:
```

Comment:
There's also a `report_failure` option for reports that you could use here: https://github.com/linkedin/datahub/blob/master/metadata-ingestion/src/datahub/ingestion/api/source.py#L27:L27. See also `report_warning`.